cmd/hegel/main.go

```diff
@@ -24,7 +30,183 @@ var (
 	customEndpoints string
 )
 
+const EnvNamePrefix = "HEGEL"
+
+type RootCommandOptions struct {
+	Facility       string `mapstructure:"facility"`
+	DataModel      string `mapstructure:"data-model"`
+	TrustedProxies string `mapstructure:"trusted-proxies"`
+
+	HTTPPort            int    `mapstructure:"http-port"`
+	HTTPCustomEndpoints string `mapstructure:"http-custom-endpoints"`
+
+	GrpcTLSCertPath string `mapstructure:"grpc-tls-cert"`
+	GrpcTLSKeyPath  string `mapstructure:"grpc-tls-key"`
+	GrpcUseTLS      bool   `mapstructure:"grpc-use-tls"`
+
+	KubeAPI    string `mapstructure:"kubernetes-api"`
+	Kubeconfig string `mapstructure:"kubeconfig"`
+}
+
+type RootCommand struct {
+	*cobra.Command
+	RootCommandOptions
+}
+
+func NewRootCommand() (*RootCommand, error) {
+	rootCmd := &RootCommand{
+		Command: &cobra.Command{
+			Use: os.Args[0],
+			Long: `
+Run a Hegel server.
+
+Each CLI argument has a corresponding environment variable in the form of the CLI argument prefixed with HEGEL. If both
+the flag and environment variable form are specified, the flag form takes precedence.
+
+Examples
+  --factility              HEGEL_FACILITY
+  --http-port              HEGEL_HTTP_PORT
+  --http-custom-endpoints  HEGEL_HTTP_CUSTOM_ENDPOINTS
+
+For backwards compatibility a set of deprecated CLI and environment variables are still supported. Behavior for
+specifying both deprecated and current forms is undefined.
+
+Deprecated CLI Flags
+  Deprecated   Current
+  --http_port  --http-port
```

Comment:
Why not [mark the flags as deprecated](https://pkg.go.dev/github.com/spf13/pflag#FlagSet.MarkDeprecated)?